### DISCUSSION TITLE
ReactiveUI.SourceGenerators forward DataAnnotation to generated properties

### DISCUSSION THREAD

**Discussion by dathoang** [reactiveui/ReactiveUI#3928] [General]
I want to add some DataAnnotation to the generated properties when using [Reactive] and [ObservableAsProperty].
Is this possible or any workaround without manually write the properties.
Eg:
``` 
[ObservableAsProperty]
[DisplayFormat(DataFormatString = "n0")]
private decimal _totalAmount;
```
Desirable outcome
```
private ObservableAsPropertyHelper<decimal> _totalAmount;

[DisplayFormat(DataFormatString = "n0")]
public decimal TotalAmount => _totalAmount.Value;
```

**Comment by ChrisPulman** [comment]
You should be able to use the following:

```csharp
[ObservableAsProperty]
[property: DisplayFormat(DataFormatString = "n0")]
private decimal _totalAmount;
```

**Reply by ChrisPulman** [reply]
Adding `property:` to the attribute specifies where it is used, this applies to [Reactive] , [ObservableAsProperty] and [ReactiveCommand] attributes
